### PR TITLE
CAMEL-11600: Implementation of a micrometer component

### PR DESCRIPTION
**Pull request by ohr**
Implementation of a micrometer component (including docs and example module), in view of Spring Boot 2.0 support in Camel 2.22 (but obviously usable outside of Spring Boot, too).

### CODE REVIEW

File: components/camel-micrometer/src/main/java/org/apache/camel/component/micrometer/eventnotifier/MicrometerEventNotifierMBean.java [hunk 1]
```diff
@@ -14,22 +14,16 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
```
Comment (oscerd) [inline components/camel-micrometer/src/main/java/org/apache/camel/component/micrometer/eventnotifier/MicrometerEventNotifierMBean.java]: Why are you moving a file of camel-metrics for a new component?

Comment (ohr) [reply components/camel-micrometer/src/main/java/org/apache/camel/component/micrometer/eventnotifier/MicrometerEventNotifierMBean.java]: * Move must have happened accidentally... I restored the original file
* Maybe http://camel.apache.org/how-do-i-add-a-component.html needs an update for occasional Camel contributors like me?
* umm, I did run -Psourcecheck before. Do you see any code style issues?

Comment (oscerd) [reply components/camel-micrometer/src/main/java/org/apache/camel/component/micrometer/eventnotifier/MicrometerEventNotifierMBean.java]: The documentation is old, we are moving to a new website and we are maintaining docs directly on github to be able to be consistent.

**Comment by oscerd** [comment]
Merged on master.

**Pull request closed without merging**